### PR TITLE
Allow running under centos 9

### PR DESCRIPTION
**Pull request by AdamSaleh**
Checked by running ./runtest.sh on Centos 7,  Centos Stream 8 and Centos Stream 9.

Bit wary that it is mostly 'works on my machine', but seems good enough for now.

### CODE REVIEW

**Comment by AlexBaranowski** [comment]
I'm not the owner of the repo, but you are mixing:
```
if (t_GetPkgRel basesystem | grep -q el9)
```
and
```
if [ "$centos_ver" -eq "9" ]; then
```
is there a reason for that?

I manage our fork at EuroLinux, and I will backport these changes, as we have just released 9 beta and partially done the same work.

IMO these patches looks neat.

You can use Vagrant to check CentOS 7 and CentOS 8. I can provide you with code.

EDIT: Grammarly + Info about Vagrant

**Comment by AdamSaleh** [comment]
No reason. I was mostly following the style those tests were written, there are more such ifdefs for other centos versions.

**Comment by AdamSaleh** [comment]
@AlexBaranowski  I have vagrant up and running but if you have partial work done, I would very much like to see your fork :-) I am new to the work on centos, and seemed to be a good way to cut my teeth in. It is not a given you should be backporting my changes and not the other way around :)

**Comment by AlexBaranowski** [comment]
@AdamSaleh I will provide it during weekend

File: tests/p_iptables/iptables_add-remove_test.sh [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 t_Log "Running $0 - verify iptables can insert and delete rules"
 
-if [ "$centos_ver" = "7" ] || [ "$centos_ver" = "8" ];then
+if [ "$centos_ver" = "7" ] || [ "$centos_ver" = "8" ] || [ "$centos_ver" = "9" ];then
```
Comment (bstinsonmhk) [inline tests/p_iptables/iptables_add-remove_test.sh]: Can we switch these to `-ge` conditionals? If versions after 7 use firewalld we can then rely on a greater-than test without changing again for CentOS Stream 10

File: tests/p_libxml2-python/0-install-libxml2-python.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@
 t_Log "Running $0 - installing libxml2-python."
 
 if [ "$centos_ver" -ge 8 ] ; then
-t_InstallPackage python36 python3-libxml2
```
Comment (bstinsonmhk) [inline tests/p_libxml2-python/0-install-libxml2-python.sh:8]: Do we need to install *some* version of python3? Does this work like this on EL8?

File: tests/p_rsync/0-install-rsync.sh [hunk 3]
```diff
@@ -3,14 +3,14 @@
 
 t_Log "Running $0 - installing rsync and xinetd."
 
-if [ "$centos_ver" = "8" ] ; then
-  t_InstallPackage xinetd rsync rsync-daemon
+if [ "$centos_ver" = "8" ] || [ "$centos_ver" = "9" ] ; then
+  t_InstallPackage rsync rsync-daemon
```
Comment (bstinsonmhk) [inline tests/p_rsync/0-install-rsync.sh:7]: `-ge 8` test here. Is xinetd not needed?

File: tests/p_rsync/0-install-rsync.sh [hunk 4]
```diff
@@ -3,14 +3,14 @@
...
 else
   t_InstallPackage xinetd rsync
 fi
 
 # Restart in case previous tests allready installed xinetd
-if [ "$centos_ver" = "7" ] || [ "$centos_ver" = "8" ]; then
+if [ "$centos_ver" = "7" ] || [ "$centos_ver" = "8" ] || [ "$centos_ver" = "9"] ; then
```
Comment (bstinsonmhk) [inline tests/p_rsync/0-install-rsync.sh]: `-ge 7` test here

**Review by bstinsonmhk — COMMENTED** [review]
This is looking good! @hughesjr can you run t_functional from this commit on your next CentOS Linux 7 and CentOS Stream 8 pushes to confirm that we haven't introduced new trouble?

File: tests/p_libxml2-python/0-install-libxml2-python.sh (hunk 2 above)
Comment (AdamSaleh) [reply tests/p_libxml2-python/0-install-libxml2-python.sh:8]: Let me check.

File: tests/p_rsync/0-install-rsync.sh (hunk 3 above)
Comment (AdamSaleh) [reply tests/p_rsync/0-install-rsync.sh:7]: No. Systemd is sufficient.

File: tests/p_iptables/iptables_add-remove_test.sh (hunk 1 above)
Comment (AdamSaleh) [reply tests/p_iptables/iptables_add-remove_test.sh]: Ok, will check for the conditionals, as I said, I mostly just used the style the tests were written in, but you are right -ge makes more sense :)

File: tests/p_libxml2-python/0-install-libxml2-python.sh (hunk 2 above)
Comment (AdamSaleh) [reply tests/p_libxml2-python/0-install-libxml2-python.sh:8]: Ok, forgot to push a commit where I reverted that, will include it when I will change those `-ge` flags.

**Comment by AdamSaleh** [comment]
Ok, I think I found all of the `"$centos_ver" = "9"` where we want to have `-ge` instead.

**Comment by hughesjr** [comment]
We need to disable z_repoclosure script on el8 (or > 7 would also work if you want it disabled on 9) .. but let me finishe the run on el7.  With repoclosure disabled, I have a good run on stream 8.

**Comment by hughesjr** [comment]
I tested in el7 and stream-8 .. I am ok with this  if we just add :

diff --git a/tests/z_repoclosure/repoclosure_test.sh b/tests/z_repoclosure/repoclosure_test.sh
index ab637d7..d19f83b 100755
--- a/tests/z_repoclosure/repoclosure_test.sh
+++ b/tests/z_repoclosure/repoclosure_test.sh
@@ -5,6 +5,8 @@ t_Log "Running $0 -  repoclosure test ."
 
 if [ $SKIP_QA_HARNESS -eq 1 ] && [ "$PRE_UPDATES" != "1" ] ; then
     t_Log "Skip this test in non QA harness environment"
+elif [ "$centos_ver" -gt "7" ] ; then
+    t_Log "Skip this test in CentOS versions greater than 7 due to modules"
 else

**Comment by AdamSaleh** [comment]
In the end I got stuck on the other tests, so I stashed all of my other changes and pushed the one you suggested :-) Plan to finish the rest of it, I just don't want this to have to wait more days to be merged.

**Pull request merged**